src/pipecat/processors/transcript_processor.py

```diff
@@ -0,0 +1,235 @@
+#
+# Copyright (c) 2024, Daily
+#
+# SPDX-License-Identifier: BSD 2-Clause License
+#
+
+from abc import ABC, abstractmethod
+from typing import List
+
+from loguru import logger
+
+from pipecat.frames.frames import (
+    ErrorFrame,
+    Frame,
+    OpenAILLMContextAssistantTimestampFrame,
+    TranscriptionFrame,
+    TranscriptionMessage,
+    TranscriptionUpdateFrame,
+)
+from pipecat.processors.aggregators.openai_llm_context import OpenAILLMContextFrame
+from pipecat.processors.frame_processor import FrameDirection, FrameProcessor
+
+
+class BaseTranscriptProcessor(FrameProcessor, ABC):
+    """Base class for processing conversation transcripts.
+
+    Provides common functionality for handling transcript messages and updates.
+    """
+
+    def __init__(self, **kwargs):
+        """Initialize processor with empty message store."""
+        super().__init__(**kwargs)
+        self._processed_messages: List[TranscriptionMessage] = []
+        self._register_event_handler("on_transcript_update")
+
+    async def _emit_update(self, messages: List[TranscriptionMessage]):
+        """Emit transcript updates for new messages.
+
+        Args:
+            messages: New messages to emit in update
+        """
+        if messages:
+            self._processed_messages.extend(messages)
+            update_frame = TranscriptionUpdateFrame(messages=messages)
+            await self._call_event_handler("on_transcript_update", update_frame)
+            await self.push_frame(update_frame)
+
+    @abstractmethod
+    async def process_frame(self, frame: Frame, direction: FrameDirection):
+        """Process incoming frames to build conversation transcript.
+
+        Args:
+            frame: Input frame to process
+            direction: Frame processing direction
+        """
+        await super().process_frame(frame, direction)
```

Comment:
we probably don't need this method since it doesn't do anything. the parent `FrameProcessor` one will be used.